pkg/package-server-manager/controller.go

```diff
@@ -102,6 +108,21 @@ func (r *PackageServerCSVReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	return ctrl.Result{}, nil
 }
 
+func ensureRBAC(client client.Client, ctx context.Context, namespace string, log logr.Logger) error {
+	log.Info("checking to see if required RBAC exists")
+	if err := client.Get(ctx, types.NamespacedName{Name: "olm-operator-serviceaccount", Namespace: namespace}, &corev1.ServiceAccount{}); err != nil {
+		return fmt.Errorf("could not get service account:%v", err)
+	}
+	if err := client.Get(ctx, types.NamespacedName{Name: "system:controller:operator-lifecycle-manager"}, &rbacv1.ClusterRole{}); err != nil {
+		return fmt.Errorf("could not get ClusterRole:% v", err)
+	}
+	if err := client.Get(ctx, types.NamespacedName{Name: "olm-operator-binding-openshift-operator-lifecycle-manager"}, &rbacv1.ClusterRoleBinding{}); err != nil {
+		return fmt.Errorf("could not get ClusterRoleBinding: %v", err)
+	}
```

Comment:
Does our cache already have ServiceAccounts, ClusterRoels and ClusterRoleBindings in it?

If not, this will likely cause our controller to cache all of them.

I'd suggest either configuring the cache to select _just_ on these particular names with a field selector or to use the direct APIReader client.

Reply:
See here: https://github.com/openshift/operator-framework-olm/blob/b3bfaa4b69bf7dfe8254cdadd8a6c8fc450e2b97/cmd/package-server-manager/main.go#L95-L104